packages/block-editor/src/components/inserter/media-tab/media-preview.js

```diff
@@ -203,7 +203,18 @@ export function MediaPreview( { media, onClick, composite, category } ) {
 	const onMouseLeave = useCallback( () => setIsHovered( false ), [] );
 	return (
 		<>
-			<InserterDraggableBlocks isEnabled={ true } blocks={ [ block ] }>
+			<InserterDraggableBlocks
+				isEnabled={ true }
+				blocks={ [ block ] }
+				onDragStart={ ( event ) => {
+					// Also set the serialized block as the HTML data for the drag event,
+					// so that it can be parsed by the media placeholder.
+					event.dataTransfer.setData(
+						'text/html',
+						serialize( [ block ] )
+					);
```

Comment:
Personally, I think this should probably be done within `InserterDraggableBlocks` and if we're adding support for multiple drag types, maybe we should add it directly into the `Draggable` component instead?

Reply:
> if we're adding support for multiple drag types, maybe we should add it directly into the Draggable component instead?

I'm afraid that would probably have a bigger impact than what I expected in this PR. We 
can definitely try doing it in a follow-up PR though.